src/Populator/ArrayPropertyMappingPopulator.php

```diff
@@ -18,52 +18,64 @@
  */
 final class ArrayPropertyMappingPopulator implements Populator
 {
-    /** @var \Closure(mixed, TContext=):mixed */
-    private \Closure $mapper;
+    private ?string $sourceArrayItemProperty;
+    /** @var \Closure(mixed, TContext=):mixed|null */
+    private ?\Closure $mapper;
     private PropertyAccessorInterface $arrayItemAccessor;
-    private PropertyAccessorInterface $accessor;
+    private PropertyMappingPopulator $populator;
 
     /**
      * @param \Closure(mixed, TContext=):mixed|null $mapper
      */
     public function __construct(
-        private string $targetProperty,
-        private string $sourceArrayProperty,
-        private ?string $sourceArrayItemProperty = null,
+        string $targetProperty,
+        string $sourceArrayProperty,
+        ?string $sourceArrayItemProperty = null,
         ?\Closure $mapper = null,
         ?PropertyAccessorInterface $arrayItemAccessor = null,
         ?PropertyAccessorInterface $accessor = null,
     ) {
-        $this->mapper = $mapper ?? static fn ($v) => $v;
+        $this->sourceArrayItemProperty = $sourceArrayItemProperty;
+        $this->mapper = $mapper;
         $this->arrayItemAccessor = $arrayItemAccessor ?? PropertyAccess::createPropertyAccessor();
-        $this->accessor = $accessor ?? PropertyAccess::createPropertyAccessor();
+        $this->populator = new PropertyMappingPopulator(
+            $targetProperty,
+            $sourceArrayProperty,
+            null,
+            $this->unwrapAndMap(...),
+            $accessor,
+        );
     }
 
     /**
      * @throws PopulationException
      */
     public function populate(object $target, object $source, ?object $ctx = null): void
     {
-        try {
-            $sourceArrayPropertyValues = $this->accessor->getValue($source, $this->sourceArrayProperty);
+        $this->populator->populate($target, $source, $ctx);
```

Comment:
Ich erkenne leider nicht, wie Du aus dem "alten" array_map einen einzelnen Populator Aufruf gemacht hast.
Wie steckt das mit dem unwrapAndMap zusammen?

Reply:
This reuses the `PropertyMappingPopulator` (as we have already done with the `ConvertingPopulator`). `unwrapAndMap` is its mapper, and contains a refactored version (with early returns) of the lines 50-58 and 63 of the old code. 

There was also previously only a single “populator call” (`$this->accessor->setValue(...)`) because it only handles one property being populated. This property contains an array that is handled via `array_map` in `unwrapAndMap`.

The advantage is that we now have all the features that the `PropertyMappingPopulator` offers, such as default values and skip null handling.